platform/kube/conversion.go

```diff
@@ -147,3 +155,111 @@ func modelToKube(km model.KindMap, k *model.Key, v proto.Message) (*Config, erro
 
 	return out, nil
 }
+
+func convertIngress(ingress v1beta1.Ingress) map[model.Key]proto.Message {
+	messages := make(map[model.Key]proto.Message)
+
+	keyOf := func(ruleNum, pathNum int) model.Key {
+		return model.Key{
+			Kind:      model.IngressRule,
+			Name:      encodeIngressRuleName(ingress.Name, ruleNum, pathNum),
+			Namespace: ingress.Namespace,
+		}
+	}
+
+	if ingress.Spec.Backend != nil {
+		messages[keyOf(0, 0)] = createIngressRule("", "", ingress.Namespace, *ingress.Spec.Backend)
+	}
+
+	for i, rule := range ingress.Spec.Rules {
+		for j, path := range rule.HTTP.Paths {
+			messages[keyOf(i+1, j+1)] = createIngressRule(rule.Host, path.Path, ingress.Namespace, path.Backend)
+		}
+	}
+
+	return messages
+}
+
+func createIngressRule(host string, path string, namespace string, backend v1beta1.IngressBackend) proto.Message {
+	destination := serviceHostname(backend.ServiceName, namespace)
+
+	rule := &config.RouteRule{
+		Destination: destination,
+		Match:       &config.MatchCondition{},
+		Route: []*config.DestinationWeight{
+			{
+				Destination: destination,
+				Weight:      100,
+
+				// A temporary measure to communicate the destination service's port
+				// to the proxy configuration generator. This can be improved by using
+				// a dedicated model object for IngressRule (instead of reusing RouteRule),
+				// which exposes the necessary target port field within the "Route" field.
+				Tags: map[string]string{
```

Comment:
That can only happen if there is only one port, and the upstream cluster becomes just the service name ("a.svc....") or service-tag combination ("a.svc...::env=prod")